src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -946,9 +948,21 @@ zocl_xclbin_load_pskernel(struct drm_zocl_dev *zdev, void *data)
 	 */
 	zocl_read_sect_kernel(AIE_RESOURCES, &aie_res, axlf, xclbin);
 
+	aienode = of_find_node_by_name(NULL, "ai_engine");
+	if (aienode == NULL)
+		DRM_WARN("AI Engine Device Node not found!");
+	else {
+		ret = of_property_read_u8(aienode, "xlnx,aie-gen", &hw_gen);
+		if (ret < 0) {
+			DRM_WARN("no aie dev generation information in device tree\n");
+		}
+		of_node_put(aienode);
```

Comment:
`aienode = NULL;`

We should reset the node to NULL because we are returning our reference?